src/web/client/WebExtensionContext.ts

```diff
@@ -381,14 +384,25 @@ class WebExtensionContext implements IWebExtensionContext {
                 schema
             );
         } catch (error) {
-            const errorMsg = (error as Error)?.message;
-            this.telemetry.sendAPIFailureTelemetry(
-                requestUrl,
-                languageEntityName,
-                Constants.httpMethod.GET,
-                new Date().getTime() - requestSentAtTime,
-                errorMsg
-            );
+            if ((error as Response)?.status>0){
+                const errorMsg = (error as Error)?.message;
+                this.telemetry.sendAPIFailureTelemetry(
+                    requestUrl,
+                    languageEntityName,
+                    Constants.httpMethod.GET,
+                    new Date().getTime() - requestSentAtTime,
+                    errorMsg,
+                    '',
+                    telemetryEventNames.WEB_EXTENSION_POPULATE_LANGUAGE_ID_TO_CODE_API_ERROR,
+                    (error as Response)?.status.toString()
+                );
+            }else{
```

Comment:
formatting - suggest keeping prettier on

Reply:
formatting in next PR